src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -498,20 +533,23 @@ Transactor::checkSingleSign(PreclaimContext const& ctx)
 {
     // Check that the value in the signing key slot is a public key.
     auto const pkSigner = ctx.tx.getSigningPubKey();
-    if (!publicKeyType(makeSlice(pkSigner)))
+    if (!(ctx.flags & tapDRY_RUN) && !publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace())
             << "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH;  // FIXME: should be better error!
     }
 
     // Look up the account.
-    auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
     auto const idAccount = ctx.tx.getAccountID(sfAccount);
     auto const sleAccount = ctx.view.read(keylet::account(idAccount));
     if (!sleAccount)
         return terNO_ACCOUNT;
 
+    // This ternary is only needed to handle `simulate`
```

Comment:
In addition to this comment, I'd like to see something like this assert

```suggestion
XRPL_ASSERT((ctx.flags & tapDRY_RUN) || !pkSigner.empty(), "ripple::Transactor::checkSingleSign : non-empty signer or simulation")
```

Reply:
Fixed in a744645f7d2e7142512791a6bb2abaaff26bd62e

Reply:
Actually, upon further examination (and failing tests), this is incorrect. The public key isn't necessary if you're running `simulate`, but if it's included, it will be checked.

Reply:
In this case it should be `(ctx.flags & tapDRY_RUN) || !pkSigner.empty()` , right ?

Reply:
Excellent point, added in 21a3341ce3220ff62b3f98da2018db3d15616499.